pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -406,6 +407,16 @@ def dump_checkpoint(self, weights_only: bool = False) -> dict:
             else:
                 checkpoint[pl.LightningModule.CHECKPOINT_HYPER_PARAMS_KEY] = dict(model.hparams)
 
+        if datamodule and datamodule.hparams:
+            if hasattr(datamodule, "_hparams_name"):
+                checkpoint[pl.LightningDataModule.CHECKPOINT_HYPER_PARAMS_NAME] = datamodule._hparams_name
```

Comment:
Shouldn't this be?

```suggestion
                checkpoint[datamodule.CHECKPOINT_HYPER_PARAMS_NAME] = datamodule._hparams_name
```

Also for `LightningModule`, in case it was modified